AFM: Docs for the April release (#29)
mulesoft/docs-functional-monitoring: modules/ROOT/pages/afm-create-monitor.adoc

include::_attributes.adoc[]
endif::[]

> You can create a monitor by filling in fields if you do not know how to write tests in the Behavior Driven Development (BDD) test-writing language or how to use the BAT CLI. This method of creating a monitor creates a test that tests one or more endpoints, and schedules the test to run at intervals that you specify.

Reviewer: To assist global comprehension:
```suggestion
You can create a monitor by completing fields if you do not know how to write tests in the Behavior Driven Development (BDD) test-writing language or how to use the BAT CLI. This method of creating a monitor creates a test that both tests one or more endpoints and schedules the test to run at intervals that you specify.
```

After every interval, you can view the results of the test and statistics about API performance.

== Before You Begin

* Determine whether you want to monitor the endpoints in one or more public APIs, one or more private APIs, or a mix of both public APIs and private APIs.
+

> For details, see xref:afm-monitoring-public-apis.adoc[Monitoring the Endpoints of Public APIs] and xref:afm-monitoring-private-apis.adoc[Monitoring the Endpoints of Private APIs and Mixes of Both Public and Private APIs].

Reviewer: ditto previous title suggestion; GLOBAL
I will stop marking this.

> * Ensure that your user ID has the "Anypoint Monitoring User" permission.

Reviewer: ```suggestion
* Ensure that your user ID has the Anypoint Monitoring User permission.
```

== Procedure
To create a monitor:

. In the *Functional Monitoring* section of Anypoint Monitoring, click *Create Monitor* in the top-right corner of your browser window.

> . In the *Monitor Setup* section, specify the name of the monitor, the location from which to run the test, how often to run the test, and whether to validate SSL certificates.

Reviewer: ```suggestion
. In the *Monitor Setup* section, provide the following information:
```
To avoid repetition

> .. Specify a name for the monitor.

Reviewer: ```suggestion
* A name for the monitor
```

> .. Select the location from which to run the test.

Reviewer: ```suggestion
* The location from which to run the test
```

+
Public locations are grouped separately from private locations in the list. You can create a private location by clicking an option in the list.

> .. Specify the intervals at which you want the monitor to run the test.

Reviewer: ```suggestion
* The interval at which you want the monitor to run the test
```

+
The interval that you select begins when you click *Create Monitor* at the end of this procedure.

> .. Toggle the *SSL Certificate Validation* switch off if the endpoints that your monitor will test use self-signed certificates.

Reviewer: ```suggestion
* The setting of the *SSL Certificate Validation* switch:
** Off prevents test failure if the endpoints that your monitor will test use self-signed certificates.
Off also permits but does not require endpoints that use SSL certificates to be signed by a trusted authority
** On requires endpoints that use SSL certificates to be signed by a trusted authority
```
Or similar. It's possible that we need only specify that off prevents self-signed failure and on requires SSL to be signed.

+
If they do use self-signed certificates and you do not toggle this switch off, tests against those endpoints fail.
+
** If the toggle is on, all endpoints that use SSL certificates must have certificates that are signed by a trusted authority.

> ** If the toggle is off, all endpoints that use SSL certificates must be self-signed.

Reviewer 2: Is not that all endpoints must be self-signed. It may or may not be

Author: @diegor , I change the bullet point to this: "If the toggle is off, endpoints that use SSL certificates may be self-signed, but do not have to be."

> ** If the toggle is off, endpoints that use SSL certificates may be self-signed, but do not have to be.

Reviewer: See above. Trying to simplify this, but now that I see the following paragraph, I don't know if that is possible.
It seems to me that this is too much conceptual information that is not part of the focus of this task, and perhaps should be covered elsewhere (about endpoint types and settings).
Or maybe we could just omit the following info about testing endpoints, as that is somewhat outside of the obligation, for THIS task, to select a switch setting.

+
To test a combination of endpoints that use trusted SSL certificates and endpoints that use self-signed certificates, create two separate monitors: one for testing the first group of endpoints, the other for testing the second group of endpoints.
. Click *Next*.
. In the *Endpoints* section, specify one or more endpoints to include in the test, the HTTP methods to use to test them, the status codes that each endpoint is expected to return, and any assertions that you want to include for each endpoint.
. Click *Next*.
. If you want to be notified of failed tests, select one or more reporters.
. Click *Create monitor*.

== Result

The monitor that you created is listed at the top of the list of your monitors. API Functional Monitoring begins counting down the interval of time that you selected. When the interval expires, the test is run.

[IMPORTANT]
====

> Tests fail when they take more than 120 seconds to run. As a workaround, you can split a long-running test into multiple tests. Download the monitor, split the test across multiple monitors in the BAT CLI, run and time each test to be sure that it is under 120 seconds, and then upload the monitors.

Reviewer: Consider steps

If you do not want to split a long-running test, download the monitor and run it from the BAT CLI.
====

> After the test is complete, two things happen:

Reviewer: Is this strictly true? Two things happen after the first test run; only one thing happens subsequently.

* The countdown of the interval begins again.

> * After the test runs for the first time, a summary of the results appears under *Endpoint* in the list entry for the monitor. The summary is something similar to "1 Failed" or "1 Failed/3 Passed", with the sum of the numbers in the summary indicating the number of endpoints in the test. Click the list entry for the monitor to view the details of the test results.

Reviewer: GLOBALLY limit the length of list items per the Style Reference:
```suggestion
* After the test runs for the first time, a summary of the results appears under *Endpoint* in the list entry for the monitor. 
The summary is something similar to "1 Failed" or "1 Failed/3 Passed", with the sum of the numbers in the summary indicating the number of endpoints in the test. You can view the details of the test results by the list entry for the monitor.
```
Avoids imperative, which implies a task within concept info.
